### PR TITLE
x509: use actual issuer name if a CA is used

### PR DESCRIPTION
**Pull request by faramir-dev**
Fixes openssl#16080.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by kaduk** [comment]
Without commenting on the actual code changes, I don't think that the string "ecdsa01RootCaCert" is the right text for the commit message.  Maybe "x509: use actual issuer name if a CA is used"?

File: apps/x509.c [hunk 1]
```diff
@@ -567,8 +567,6 @@ int x509_main(int argc, char **argv)
             goto end;
         }
 
-        if (!X509_set_issuer_name(x, X509_REQ_get_subject_name(req)))
-            goto end;
```
Comment (t8m) [inline apps/x509.c:571]: Keep this call as is, please, as in case `-CA` is not used there will be no issuer set.

File: apps/x509.c [hunk 2]
```diff
@@ -591,6 +589,9 @@ int x509_main(int argc, char **argv)
         if (xca == NULL)
             goto end;
     }
+    issuer_cert = xca != NULL ? xca : x;
+    if (!X509_set_issuer_name(x, X509_get_subject_name(issuer_cert)))
+        goto end;
```
Comment (t8m) [inline apps/x509.c:594]: Just move this inside the above if and you do not need to have the issuer_cert variable.

**Review by t8m — CHANGES_REQUESTED** [review]

**Review by kaduk — APPROVED** [review]
This codebase is structured quite poorly for maintainability...but it looks like this change should be okay.

**Comment by t8m** [comment]
Merged to 1.1.1 branch. Thank you.

**Pull request closed without merging**